src/core/xfa/bind.js

```diff
@@ -637,6 +637,9 @@ class Binder {
           if (!match) {
             // We're in matchTemplate mode so create a node in data to reflect
             // what we've in template.
+            if (child.occur.min === 0) {
+              uselessNodes.push(child);
```

Comment:
- The comment here doesn't apply to the added code so you should move it
- you should add a comment (you can quote the specification or add a link to it) to explain why the node is useless.
- once we know that the child is useless, you can then skip the code after.

Reply:
👍🏽